pkg/apis/podgroup/v1alpha1/types.go

```diff
@@ -0,0 +1,134 @@
+	PodGroupScheduling PodGroupPhase = "Scheduling"
+
+	// PodScheduled means `spec.minMember` pods of PodGroups have been scheduled finished and pods have been in running
+	// phase.
+	PodGroupScheduled PodGroupPhase = "Scheduled"
+
+	// PodGroupUnknown means part of `spec.minMember` pods are running but the other part can not
+	// be scheduled, e.g. not enough resource; scheduler will wait for related controller to recover it.
+	PodGroupUnknown PodGroupPhase = "Unknown"
+
+	// PodGroupFinish means all of `spec.minMember` pods are successfully.
+	PodGroupFinished PodGroupPhase = "Finished"
+
+	// PodGroupFailed means at least one of `spec.minMember` pods is failed.
+	PodGroupFailed PodGroupPhase = "Failed"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodGroup is a collection of Pod; used for batch workload.
+type PodGroup struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the pod group.
+	// +optional
+	Spec PodGroupSpec `json:"spec,omitempty"`
+
+	// Status represents the current information about a pod group.
+	// This data may not be up to date.
+	// +optional
+	Status PodGroupStatus `json:"status,omitempty"`
+}
+
+// PodGroupSpec represents the template of a pod group.
+type PodGroupSpec struct {
+	// MinMember defines the minimal number of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	MinMember uint32 `json:"minMember,omitempty"`
+
+	// MinResources defines the minimal resource of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	MinResources *v1.ResourceList `json:"minResources,omitempty"`
+
+	// MaxScheduleTime defines the maximal time of members/tasks to wait before run the pod group;
+	MaxScheduleTime *metav1.Duration `json:"maxScheduleTime,omitempty"`
+}
+
+// PodGroupStatus represents the current state of a pod group.
+type PodGroupStatus struct {
+	// Current phase of PodGroup.
+	Phase PodGroupPhase `json:"phase,omitempty"`
+
+	// OccupiedBy marks the workload (e.g., deployment, statefulset) UID that occupy the podgroup.
+	// It is empty if not initialized.
+	OccupiedBy string `json:"occupiedBy,omitempty"`
+
+	// The number of actively running pods.
+	// +optional
+	Scheduled uint32 `json:"scheduled,omitempty"`
+
+	// The number of actively running pods.
+	// +optional
+	Running uint32 `json:"running,omitempty"`
+
+	// The number of pods which reached phase Succeeded.
+	// +optional
+	Succeeded uint32 `json:"succeeded,omitempty"`
+
+	// The number of pods which reached phase Failed.
+	// +optional
+	Failed uint32 `json:"failed,omitempty"`
+
+	// ScheduleStartTime of the group
+	ScheduleStartTime metav1.Time `json:"scheduleStartTime,omitempty"`
```

Comment:
ScheduleStartTime is the create time of PG ? 
If So, it's better to rename it to CreateTime?

Reply:
This is field just record when the group starts scheduling

Reply:
Thanks, It also need to add the field to record the create time of PG.

Reply:
MetaData of CRD already includes this field!

Reply:
https://github.com/kubernetes-sigs/scheduler-plugins/pull/52/files#diff-23885a062c430d62621c160ab8aaaf08R66

Reply:
Ok, Thanks